packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,225 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <UMCore/UMAppLifecycleService.h>
+#import <UMCore/UMDefines.h>
+#import <UMCore/UMEventEmitterService.h>
+#import <UMCore/UMModuleRegistryProvider.h>
+
+#import <EXScreenOrientation/EXScreenOrientationModule.h>
+#import <EXScreenOrientation/EXScreenOrientationUtilities.h>
+#import <EXScreenOrientation/EXScreenOrientationRegistry.h>
+
+#import <UIKit/UIKit.h>
+
+@interface EXScreenOrientationModule ()
+
+@property (nonatomic, weak) EXScreenOrientationRegistry *screenOrienationRegistry;
+@property (nonatomic, weak) id<UMEventEmitterService> eventEmitter;
+@property (nonatomic, assign) bool hasListeners;
+@property (nonatomic, assign) UIInterfaceOrientation currentScreenOrientation;
+
+@end
+
+@implementation EXScreenOrientationModule
+
+UM_EXPORT_MODULE(ExpoScreenOrientation);
+
+- (void)dealloc
+{
+  [self stopObserving];
+  
+  dispatch_sync(dispatch_get_main_queue(), ^{
+    [[UIDevice currentDevice] endGeneratingDeviceOrientationNotifications];
+  });
+}
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _eventEmitter = [moduleRegistry getModuleImplementingProtocol:@protocol(UMEventEmitterService)];
+  _screenOrienationRegistry = [moduleRegistry getSingletonModuleForName:@"EXScreenOrientationRegistry"];
+  id<UMAppLifecycleService> lifercycleService = [moduleRegistry getModuleImplementingProtocol:@protocol(UMAppLifecycleService)];
+
+  if (lifercycleService) {
+    [lifercycleService registerAppLifecycleListener:self];
+  }
+  
+  _hasListeners = NO;
+  
+  dispatch_async(dispatch_get_main_queue(), ^{
+    if (![[UIDevice currentDevice] isGeneratingDeviceOrientationNotifications]){
```

Comment:
We should nest calls to `-(void)beginGeneratingDeviceOrientationNotifications`, otherwise it could be that:
- other module begins generating
- we instantiate, we check if device is generating notifications, it is, so we don't ask to generate
- other module ends generating
- we end up not receiving notifications.

To achieve what you're doing here I think we may want to have an instance variable that would say whether this instance has already began generating or not.

+ please add space in between `){`